### PR TITLE
Retrying store

### PR DESCRIPTION
**Pull request by dgem**
Fix so that update / create operations that fail due to a neo transient error can be retried.
Also includes a couple of cosmetic changes like improving the details of the error log is a purge fails & logging all cypher queries at silly level (so they can be more easily switched off in prod)

### CODE REVIEW

File: server/v3/persistence/store.js [hunk 1]
```diff
@@ -307,12 +309,40 @@ class Store {
...
+			logger.silly({event: 'Sleep', pause,});
+			return sleep(pause)
+			.then(()=>{
+				let nappedFor = Date.now() - napStart;
+				logger.silly({event: 'Sleep', nappedFor});
+				return execute(statement, retries)
+			});
+		} else {
+			logger.error({error, retries});
+			Promise.reject(error);
```
Comment (phamann) [inline server/v3/persistence/store.js]: This needs to be returned

File: server/v3/persistence/store.js [hunk 2]
```diff
@@ -307,12 +309,40 @@ class Store {
...
 			});
 	}
 
 }
 
+function sleep (time) {
+  return new Promise((resolve) => setTimeout(resolve, time));
+}
+
+function execute(statement, retries) {
```
Comment (phamann) [inline server/v3/persistence/store.js]: You declare `retires` as a parameter, but none of your invocations of `execute` above pass in a retry value. Which is ok, but you need to assign a default value of the `MAX_RETRIES` const?

File: server/v2/lib/purge-cache.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ function purgeService(serviceName) {
 			method: 'POST'
 		}).then(res => {
 			if (!res.ok) {
-				logger.error(new Error(`Fastly purge failed with status ${res.status}`, {
+				logger.error(new Error(`Fastly purge failed with status ${res.status} for ${serviceName} with ${service.id}`, {
```
Comment (phamann) [inline server/v2/lib/purge-cache.js]: Is this meant to be in the PR?

**Comment by dgem** [comment]
I think I'm going to put together a clean branch and resubmit...

**Pull request closed without merging**